ModularTegustation/ego_weapons/_ego_weapon.dm

```diff
@@ -99,90 +100,98 @@
 
 /obj/item/ego_weapon/examine(mob/user)
 	. = ..()
-	. += EgoAttackInfo(user)
+	if(!is_ranged)
+		. += EgoAttackInfo(user)
 	if(special)
 		. += span_notice("[special]")
 	if(LAZYLEN(attribute_requirements))
-		. += span_notice("It has <a href='?src=[REF(src)];list_attributes=1'>certain requirements</a> for the wearer.")
+		if(!ishuman(user))	//You get a notice if you are a ghost or otherwise
+			. += span_notice("It has <a href='?src=[REF(src)];list_attributes=1'>certain requirements</a> for the wearer.")
+		else if(CanUseEgo(user))	//It's green if you can use it
+			. += span_nicegreen("It has <a href='?src=[REF(src)];list_attributes=1'>certain requirements</a> for the wearer.")
+		else				//and red if you cannot use it
+			. += span_danger("You cannot use this EGO!")
+			. += span_danger("It has <a href='?src=[REF(src)];list_attributes=1'>certain requirements</a> for the wearer.")
 
-	if(type in GLOB.small_ego)
-		. += span_notice("This weapon fits in an EGO belt.")
+	var/list/typecache_small = typecacheof(GLOB.small_ego)
+	if(is_type_in_typecache(src, typecache_small))
+		. += span_nicegreen("This weapon fits in an EGO belt.")
 
-	if(reach>1)
-		. += span_notice("This weapon has a reach of [reach].")
+	//Melee stuff is NOT shown on ranged lol
+	if(!is_ranged)
```

Comment:
this should just be a `if(is_ranged){return}`

Reply:
Bug fixed. I do tend to swap the two somewhat often